File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/RoundStartedEvent.java
package dev.robocode.tankroyale.botapi.events;

/** Event occurring when a new round has just started. */
@SuppressWarnings("unused")
public final class RoundStartedEvent extends BotEvent {

  /** The round number. */
  private final int roundNumber;

  /**
   * Initializes a new instance of the RoundStartedEvent class.
   *
   * @param roundNumber is the round number.
   * @param turnNumber is the turn number.
   */
  public RoundStartedEvent(int roundNumber, int turnNumber) {
    super(turnNumber);
    this.roundNumber = roundNumber;
  }

  /**
   * Returns the round number.
   *
   * @return The round number.
   */
  public int getRoundNumber() {
    return roundNumber;
  }
}

File: bot-api/dotnet/api/src/events/RoundStartedEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when a round has just started.
/// </summary>
public sealed class RoundStartedEvent : IEvent
{
    /// <summary>The round number.</summary>
    public int RoundNumber { get; }

    /// <summary>
    /// Initializes a new instance of the RoundStartedEvent class.
    /// </summary>
    /// <param name="roundNumber">The round number</param>
    [JsonConstructor]
    public RoundStartedEvent(int roundNumber) => RoundNumber = roundNumber;
}